README.md

```diff
@@ -169,12 +166,37 @@ The default behaviour is to ignore the error.
 
 #### Tagging support
 
-Tags or dimensions are not covered by the StatsD specification. Providers supporting tags have implemented their flavours:
+Tags or dimensions are not covered by the StatsD specification. Providers supporting tags have implemented their flavours. Some of the major providers are supported out-of-the-box:
 
-* [DataDog](https://docs.datadoghq.com/developers/dogstatsd/datagram_shell/?tab=metrics) and [Splunk](https://docs.splunk.com/Documentation/Splunk/8.0.5/Metrics/GetMetricsInStatsd).
+* [CloudWatch](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/CloudWatch-Agent-custom-metrics-statsd.html).
+* [DataDog](https://docs.datadoghq.com/developers/dogstatsd/datagram_shell/?tab=metrics)
 * [InfluxDB](https://www.influxdata.com/blog/getting-started-with-sending-statsd-metrics-to-telegraf-influxdb/#introducing-influx-statsd).
 * [Librato](https://github.com/librato/statsd-librato-backend#tags).
 * [SignalFX](https://docs.signalfx.com/en/latest/integrations/agent/monitors/collectd-statsd.html#adding-dimensions-to-statsd-metrics).
+* [Splunk](https://docs.splunk.com/Documentation/Splunk/8.0.5/Metrics/GetMetricsInStatsd).
+
+```csharp
+var config = new StatsDConfiguration
+{
+    Prefix = "prefix",
+    Host = "127.0.0.1",
+    TagsFormatter = SupportedTagsFormatter.CloudWatch,
+};
+```
+
+##### Extending tags formatter
+
+As tags are not part of the StatsD specification, the `IStatsDTagsFormatter` used can be extended and injected in the `StatsDConfiguration`.
+
+The template class `StatsDTagsFormatter` can be inherited providing:
+
+* **prefix**: the string that will appear before the tag(s).
+* **suffix**: the string that will appear after the tag(s).
+* **areBucketNameTags**: a boolean indicating if the tag(s) are placed right after the bucket name (like it is supported by InfluxDB, Librato or SignalFX) or otherwise at the end of the StatsD message (like it is supported by AWS CloudWatch, DataDog or Splunk).
+* **tagsSeparator**: the string that will be placed between tags.
+* **keyValueSeparator**: the string that will be placed between the tag key and its value.
+
+A new implementation of `IStatsDTagsFormatter` can be written from scratch. For this purpose there are available extension methods to write in the `Buffer`.
```

Comment:
I guess this sentence can be dropped with the refactor to hide away the buffers?